### PR TITLE
Bug fixed and added new script feature

### PR DESCRIPTION
**Pull request by Emily82**
- Bugs fixed in shell.
- Implemented shell history circular retrieval.
- Added support for downkey arrow in shell.
- Added bochsconfig script and --config bochs option
  for start.sh to add romimagebios and vgaromimagebios paths in
  .bochsrc.
- Updated documentation (about this new script feature).
- Minor cleaning

### CODE REVIEW

**Comment by amanuel2** [comment]
I would recommend to start use #include<stdint.h> and #include<stddef.h> to get types like `uint32_t` etc... rather than use naked types such as `unsigned int` etc...

**Comment by Emily82** [comment]
You are right, i know about this issue, i will have to discuss about this change with inuyasha

File: tasks/task_utils.c [hunk 1]
```diff
@@ -47,15 +49,22 @@ table_address_t map_kernel(){
...
+		map_address(pagetable, logic_pt_address);
+        //We pass to the next pagedir and pagetable
+		pagedir+=4096;//4K for each pagedir structure
+		pagetable+=4096;//4K for each pagetable structure
 		i++;
-	}	
+	}
 	//set_pagedir_entry_ric(0, (unsigned int)current_page_table, PD_PRESENT|SUPERVISOR|WRITE,0);  
 	/**Initialize the pagedir with all 0's and the pagetable with the locations of the kernel*/  
+    /*NOTE: We incremented pagedir and pagetable in the last loop we should reinitialize both with the starting values?*/
```
Comment (inuyasha82) [inline tasks/task_utils.c]: If this is an open question, and you don't get an answer this shouldn't go into a pull request. So please remove this part and we can discuss it.

**Comment by inuyasha82** [comment]
Some comments
1. Don't put temporary code (look in task_utils.c, your comment contains a question, that it means that it is not discussed yet, in a pull request you should put only the code that are 100% ready to merge.
2. Ideally a good practice is to create a pull request for every single feature implemented (if the new code add some bugs, it will be easied to identify and they are easier to read :) ), so in theory you should have created at least one pull request for the shell history, and one for the bochsconfig option.

File: tasks/task_utils.c (hunk 1 above)
Comment (Emily82) [inline tasks/task_utils.c]: Yea it was an old comment, i removed it in my New Branch and i forgot to do it for this pull.
Ok ill change all

**Comment by Emily82** [comment]
Ok, single feature -> single pull request, i will remember this important rule for next pulls.

**Comment by inuyasha82** [comment]
There is an issue in the history feature, if you go back to a specific command, and then delete the current selection using backspace, the command is deleted from the screen, but not deleted from the buffer, and if you press enter that command is still executed.

**Comment by Emily82** [comment]
Strange, i cannot reproduce the bug. If i delete completely a command from history with backspace i got this:

root~:/root#
root~:/root#

All seems to work

**Pull request merged**